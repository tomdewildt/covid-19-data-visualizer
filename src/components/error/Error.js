import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import { ReactComponent as Alert } from "../../assets/alert.svg";

const ErrorWrapper = styled.div`
    display: flex;
    width: 100vw;
    height: 100vh;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    background: ${ ( { theme } ) => theme.colors.black };
`;

const AlertIcon = styled( Alert )`
    height: 3rem;
    overflow: visible;
    stroke: ${ ( { theme } ) => theme.colors.greyLight };

    polygon {
        transform-origin: center;
        animation: ripple 3s infinite;
    }
`;

const ErrorMessage = styled.p`
    margin: ${ ( { theme } ) => theme.dimens.spacing[ 4 ] };
    color: ${ ( { theme } ) => theme.colors.greyLight };
`;

const Error = ( { title, message } ) => (
    <ErrorWrapper>
        <AlertIcon title={ title } />
        <ErrorMessage>{ message }</ErrorMessage>
    </ErrorWrapper>
);

Error.propTypes = {
    title: PropTypes.string,
    message: PropTypes.string,
};

Error.defaultProps = {
    title: "error icon",
    message: "An unexpected error occurred.",
};

export default Error;
