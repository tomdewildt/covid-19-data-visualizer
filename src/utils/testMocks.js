/* eslint-disable react/prop-types */
import React from "react";

// Components
jest.mock( "@deck.gl/react", () => ( { children } ) => (
    <mock-deckgl>{ children }</mock-deckgl>
) );

// Assets
jest.mock( "../assets/gemeenten.json", () => ( [
    {
        type: "Feature",
        properties: { Gemeente: "Gemeente 1", Gemeentecode: 1, Coordinaten: [] },
        geometry: { type: "MultiPolygon", coordinates: [] },
    },
    {
        type: "Feature",
        properties: { Gemeente: "Gemeente 2", Gemeentecode: 2, Coordinaten: [] },
        geometry: { type: "MultiPolygon", coordinates: [] },
    },
    {
        type: "Feature",
        properties: { Gemeente: "Gemeente 3", Gemeentecode: 3, Coordinaten: [] },
        geometry: { type: "MultiPolygon", coordinates: [] },
    },
] ) );

