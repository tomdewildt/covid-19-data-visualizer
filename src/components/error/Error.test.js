import React from "react";
import { axe } from "jest-axe";

import { renderWithTheme } from "../../utils/testUtils";

import Error from "./Error";

describe( "Error", () => {
    it( "renders consistently", () => {
        const { asFragment } = renderWithTheme( <Error /> );
        const fragment = asFragment();

        expect( fragment ).toMatchSnapshot();
    } );

    it( "should have no axe violations", async () => {
        const { container } = renderWithTheme( <Error /> );
        const result = await axe( container );

        expect( result ).toHaveNoViolations();
    } );

    it( "respects the title prop", () => {
        const { getByTitle } = renderWithTheme( <Error title="title" /> );
        const title = getByTitle( "title" );

        expect( title ).toBeInTheDocument();
    } );

    it( "respects the message prop", () => {
        const { getByText } = renderWithTheme( <Error message="message" /> );
        const message = getByText( "message" );

        expect( message ).toBeInTheDocument();
    } );
} );
