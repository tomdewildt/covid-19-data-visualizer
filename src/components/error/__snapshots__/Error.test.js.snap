// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders consistently 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100vw;
  height: 100vh;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #191a1a;
}

.c1 {
  height: 3rem;
  overflow: visible;
  stroke: #c4c4c4;
}

.c1 polygon {
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-animation: ripple 3s infinite;
  animation: ripple 3s infinite;
}

.c2 {
  margin: 1.5rem;
  color: #c4c4c4;
}

<div
    class="c0"
  >
    <svg
      class="c1"
      title="error icon"
    >
      alert.svg
    </svg>
    <p
      class="c2"
    >
      An unexpected error occurred.
    </p>
  </div>
</DocumentFragment>
`;
